pkg/cmd/deletecmd/delete_application.go

```diff
@@ -139,6 +139,28 @@ func (o *DeleteApplicationOptions) Run() error {
 	return nil
 }
 
+func matchesWithAnyEnvironmentURL(s string, environments map[string]*v1.Environment) bool {
```

Comment:
This looks good to me now in terms of using the "filtering" approach. Still trying to figure out whether there is no better way, aka aligning with `get application`.  Funny enough `GetDeployments` in `get application` takes a completely different approach to what we do here. `GetDeployments` goes via Kubernetes deployments while here we look at the Prow config. IMO this should definitely be aligned. Also, the code in `get applications` is explicitly skipping the dev environment which makes sense as well. So I am wondering whether we could not share somehow the code around `GetApplicationsOptions#getAppData`. Does this make sense?

Reply:
@hferentschik yup I know what you mean, I'd be ok using the approach mentioned in the other comment but we should have it clear how much backwards compatibility this should have. If it isn't necessary to show the owner for example then we could also just do the switch.

Reply:
In the light of https://codetree.com/projects/StX1/board/10487910?type=epic and the recent discussion around applications, I think applications should be retrieved via iterating SourceRepository CRDs. Initially, we can do this without any modifications to the CRD itself, we can just apply the same filtering for env repos as we use here. Another benefit of using the CRDs is that you have the repo information. 

I leave it up to you. I guess we could merge this now and address everything else as part of the "refactor application commands" or we take a first initial step and use SourceRepository CRDs. WDYT?

Reply:
@hferentschik because of all the changes and back and forth that might happen from the refactoring I'd prefer to just merge/close this PR and address in https://github.com/jenkins-x/jx/issues/6101

Reply:
We took the refactoring of the application commands into the new sprint, if you want to continue there. The first step would be to let `get` and `delete` application use the same data source (SourceRepository).